pkg/plugins/runtime/k8s/controllers/pod_converter.go

```diff
@@ -182,6 +192,61 @@ func GatewayFor(clusterName string, pod *kube_core.Pod, services []*kube_core.Se
 	}, nil
 }
 
+// DeploymentFor returns the name of the deployment that the pod exists within. The second return
+// value indicates whether or not the deployment was found when no error occurs, otherwise an
+// error is returned as the third return value.
+func (p *PodConverter) DeploymentFor(ctx context.Context, namespace string, pod *kube_core.Pod) (string, bool, error) {
+	owners := pod.GetObjectMeta().GetOwnerReferences()
+	rs := &kube_apps.ReplicaSet{}
+	for _, owner := range owners {
+		if owner.Kind == "ReplicaSet" {
+			rsKey := kube_client.ObjectKey{Namespace: namespace, Name: owner.Name}
+			if err := p.ReplicaSetGetter.Get(ctx, rsKey, rs); err != nil {
+				return "", false, err
+			}
+			break
+		}
+	}
+
+	if rs == nil {
+		return "", false, nil
+	}
+
+	rsOwners := rs.GetObjectMeta().GetOwnerReferences()
+	dpl := &kube_apps.Deployment{}
+	for _, owner := range rsOwners {
+		if owner.Kind == "Deployment" {
+			dplKey := kube_client.ObjectKey{Namespace: namespace, Name: owner.Name}
+			if err := p.DeploymentGetter.Get(ctx, dplKey, dpl); err != nil {
```

Comment:
I think you can just use the name from the owner reference, no need to get the `Deployment`.

Reply:
Yeah! It's right there, thanks for pointing that out.